### PR TITLE
Client: configurable timeouts and retry intervals

### PR DESCRIPTION
**Pull request by harbu**
Add timeouts and retry intervals as configuration options in `StrictStreamrClientConfig`. Increase timeout of polling of The Graph in production (10 seconds was not enough)

### CODE REVIEW

File: packages/client/src/ConfigBase.ts [hunk 1]
```diff
@@ -22,6 +22,22 @@ export type CacheConfig = {
     maxAge: number
 }
 
+export type TimeoutsConfig = {
```
Comment (harbu) [inline packages/client/src/ConfigBase.ts]: Don't make part of public API, keep as internal info in streamr-client.

File: packages/client/src/ConfigBase.ts [hunk 2]
```diff
@@ -78,6 +94,7 @@ export type StrictStreamrClientConfig = {
     ensCacheChainAddress: EthereumAddress,
     dataUnion: DataUnionConfig
     cache: CacheConfig,
+    timeouts: TimeoutsConfig
```
Comment (harbu) [inline packages/client/src/ConfigBase.ts]: Could `@internal` work here?

File: packages/client/src/utils/HttpFetcher.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { inject, Lifecycle, scoped } from 'tsyringe'
+import { Debugger } from 'debug'
+import { Context } from './Context'
+import { Config, StrictStreamrClientConfig } from '../Config'
+import { instanceId } from './index'
+import fetch, { Response } from 'node-fetch'
+
+@scoped(Lifecycle.ContainerScoped)
+export class HttpFetcher {
```
Comment (harbu) [inline packages/client/src/utils/HttpFetcher.ts:9]: Replace usages with `Rest#get` and rm this class?

**Review by harbu — COMMENTED** [review]
Teo's review points

File: packages/client/src/ConfigBase.ts (hunk 1 above)
Comment (harbu) [reply packages/client/src/ConfigBase.ts]: Works

File: packages/client/src/ConfigBase.ts (hunk 2 above)
Comment (harbu) [reply packages/client/src/ConfigBase.ts]: Works

File: packages/client/src/utils/HttpFetcher.ts (hunk 3 above)
Comment (harbu) [reply packages/client/src/utils/HttpFetcher.ts:9]: Actually looking at `Rest` more closely, it is design to handle session expiration with Core API requests. So not really general enough for this use case.

**Review by teogeb — APPROVED** [review]
LGTM. Good to have the timeouts configurable. Let's restructure the configs in another PR (as the `_` suffix suggest, this is just a temporary location for the timeout settings),

**Pull request merged**